pkg/generator.go

```diff
@@ -104,7 +104,10 @@ func (g *Generator) addPackageImportWithName(ctx context.Context, path, name str
 
 func (g *Generator) getNonConflictingName(path, name string) string {
 	if !g.importNameExists(name) {
-		return name
+		// do not allow imports with the same name as the package when inPackage
+		if !g.InPackage || g.iface.Pkg.Name() != name {
+			return name
```

Comment:
Let's collapse this if statement into one

```go
if !g.importNameExists(name) && (!g.InPackage || g.iface.Pkg.Name() != name)
```

Reply:
Done